File: languages/python/bloock/entity/identity_v2/verification_receipt.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto


class VerificationReceipt:
    def __init__(self, session_id: int, verification_request: str) -> None:
        self.session_id = session_id
        self.verification_request = verification_request

    @staticmethod
    def from_proto(c: proto.VerificationReceipt) -> VerificationReceipt:
        return VerificationReceipt(
            session_id=c.session_id,
            verification_request=c.verification_request,
        )

    def to_proto(self) -> proto.VerificationReceipt:
        return proto.VerificationReceipt(
            session_id=self.session_id,
            verification_request=self.verification_request,
        )

File: languages/go/entity/identityV2/verification_receipt.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type VerificationReceipt struct {
	SessionID int64
	VerificationRequest string
}

func NewVerificationReceiptFromProto(s *proto.VerificationReceipt) VerificationReceipt {
	if s == nil {
		return VerificationReceipt{}
	}
	return VerificationReceipt{
		SessionID: s.SessionId,
		VerificationRequest: s.VerificationRequest,
	}
}

func (i VerificationReceipt) ToProto() *proto.VerificationReceipt {
	return &proto.VerificationReceipt{
		SessionId: i.SessionID,
		VerificationRequest: i.VerificationRequest,
	}
}

